dev/release/verify-release-candidate.sh

```diff
@@ -217,6 +217,7 @@ test_cmake_project() {
   show_info "Configure CMake Project"
   ${CMAKE_BIN} "${NANOARROW_SOURCE_DIR}/${2}" \
     "${@:3}" \
+    -DCMAKE_POSITION_INDEPENDENT_CODE=ON \
```

Comment:
This works of course but out of interest what was the issue with the target version? It reads like you set the flags directly but you can also apply this option on a target basis and let cmake add the flags: `set_property(TARGET fpic-test PROPERTY POSITION_INDEPENDENT_CODE ON)`